### DISCUSSION TITLE
Garbage collection

### DISCUSSION THREAD

**Discussion by linglejack06** [team401/2025-Robot-Code#8] [Ideas]
Has anyone seen the way Advantage kit updated their template projects to help garbage collection on rio 1 and 2, might help our ram usage issues from 2024

```
frcJava(getArtifactTypeClass('FRCJavaArtifact')) {
    jvmArgs.add("-XX:+UnlockExperimentalVMOptions")
    jvmArgs.add("-XX:GCTimeRatio=5")
    jvmArgs.add("-XX:+UseSerialGC")
    jvmArgs.add("-XX:MaxGCPauseMillis=50")

    // The options below may improve performance, but should only be enabled on the RIO 2
    //
    // final MAX_JAVA_HEAP_SIZE_MB = 100;
    // jvmArgs.add("-Xmx" + MAX_JAVA_HEAP_SIZE_MB + "M")
    // jvmArgs.add("-Xms" + MAX_JAVA_HEAP_SIZE_MB + "M")
    // jvmArgs.add("-XX:+AlwaysPreTouch")
}

``` 

found here: https://www.chiefdelphi.com/t/advantagekit-2025-log-replay-streamlined/476019/37

@jkleiber @aidnem @PChild

**Comment by jkleiber** [comment]
@linglejack06 it's worth making a ticket in 2025-Robot-Code (and potentially in coppercore) to get these improvements

**Reply by linglejack06** [reply]
ive made tickets, likely ill solve the ticket with the ak-vision branch in coppercore. and once im done there I can get ticket solved in 2025 robot code and make a pull request there.
